File: cellulosesz-api/src/main/java/top/likoslupus/cellulosesz/api/playerstate/VanishService.java
package top.likoslupus.cellulosesz.api.playerstate;

import top.likoslupus.cellulosesz.api.admin.AdminResult;
import top.likoslupus.cellulosesz.api.platform.CellPlayer;

import java.util.UUID;

public interface VanishService {

    boolean vanished(UUID uuid);

    AdminResult setVanished(CellPlayer player, boolean vanished);

    boolean canSee(CellPlayer viewer, UUID target);

    void synchronizeViewer(CellPlayer viewer);

}

File: cellulosesz-api/src/main/kotlin/top/likoslupus/cellulosesz/api/playerstate/VanishService.kt
package top.likoslupus.cellulosesz.api.playerstate

import top.likoslupus.cellulosesz.api.platform.CellPlayer
import java.util.*
import java.util.concurrent.CompletableFuture

public interface VanishService {

    public fun vanished(uuid: UUID): Boolean

    public fun setVanished(
        player: CellPlayer,
        vanished: Boolean
    ): CompletableFuture<PlayerStateResult>

    public fun canSee(viewer: CellPlayer, target: UUID): Boolean

    public fun synchronizeViewer(viewer: CellPlayer)

}
